onnxruntime/core/framework/model_metadef_id_generator.cc

```diff
@@ -28,7 +28,7 @@ int ModelMetadefIdGenerator::GenerateId(const onnxruntime::GraphViewer& graph_vi
   // hash the bytes in the Graph instance. we can't just use the address as a new Graph instance may use
   // the same memory (unit tests prove this can occur). the raw bytes of the Graph instance should be a unique
   // fingerprint for the instance that can use used as the key to the hash of the model path/contents.
-  MurmurHash3::x86_128(&main_graph, gsl::narrow_cast<int32_t>(sizeof(Graph)), instance_hash[0], &instance_hash);
+  MurmurHash3::x86_128(&main_graph, gsl::narrow_cast<long>(sizeof(Graph)), instance_hash[0], &instance_hash);
```

Comment:
Can we change the MurmurHash3::x86_128 method to take a size_t type? Then these casts go away as it looks like everything being passed to it is size_t and it taking an 'int' is the problem.

Reply:
See above.

Reply:
I agree that the len is better to be size_t. Then we only need static_cast once in computing nBlocks.

Reply:
sure